icechunk/src/error.rs

```diff
@@ -0,0 +1,45 @@
+use std::fmt::Display;
+
+use tracing_error::SpanTrace;
+
+#[derive(Debug)]
```

Comment:
New "base" for all rich Icechunk errors. We store a cause error and the span where it happened. The spans is similar to a stack trace, but more semantic, not all functions are in the list, and they can maintain more information, such as their arguments and arbitrary extra data.